### DISCUSSION TITLE
default user/user_password cache somewhere

### DISCUSSION THREAD

**Discussion by andrelec1** [rabbitmq/rabbitmq-server#3174] [Other] [answered]
I just setup a rabbitMQ server ...
I try to change de default user login/password ...

After some test seem the environement variable in doker-compose not working ... but work with docker ...
( there are a old issus ( i can't refound ) where some-one explaine this, but he comment/close the issue saying it start working without knowing why )  ....

So after some try ... if docker compose re-use the container, after you change the value, you can see the new value prompte by entrypoint.sh , but when you list user from `rabbitmqctl` you can seed le old one  ...

The quickest / simplest fix i found is juste force docker-compose to recreate another container with the `-V`

![image](https://user-images.githubusercontent.com/3273204/124289430-9a739780-db52-11eb-800b-c3a8c59945d1.png)

I don't if this is a real issus !? but i think this can be save some hours to some one else ...

**Comment by michaelklishin** [comment, marked as answer]
RabbitMQ does not support configuration of default credentials using environment variables. It's a Docker image invention and it will
be [removed very soon](https://github.com/docker-library/rabbitmq/pull/467).

[Pre-seeding Users](https://www.rabbitmq.com/access-control.html#seeding) explains how to seed users. If this configuration
is applied after the default `guest` user was created, it **will not** be deleted. Only nodes that start for the first time seed users
(but such nodes can be reset).

[RabbitMQ Configuration guide](https://www.rabbitmq.com/configure.html) explains how to inspect effective config file location and effective node configuration.

**Reply by andrelec1** [reply]
I'm not agress with you... you can override the default user/password ( and this is what the entrypoint.sh do with the docker environnement variable ...

if i mount a `default_user.conf` in `/etc/rabbitmq/conf.d` there are same "fake" issue because default user/login only read at the first start ...

I think the only thing that can be donne is juste making user more aware about this... and if they want to change the password after, juste add `-V` for forcing docker-compose to recreate the container  and not reusing the old one ...

**Reply by michaelklishin** [reply]
i don't understand what this "fake issue" is. Like I said, the ability to override default user credentials using environment variables is going away in the Docker image very soon. See for yourself in https://github.com/docker-library/rabbitmq/pull/467

**Comment by michaelklishin** [comment]
I also would never recommend enabling remote access for user `guest`, which is what your screenshot suggests is done in the generated `rabbitmq.conf`.

**Reply by andrelec1** [reply]
No, if you override the default user, guest didn't exist anymore ...

**Reply by michaelklishin** [reply]
Then the loopback users line can be removed.

**Comment by andrelec1** [comment]
I think printing  default user/password at start in log isn't good idea too !

**Reply by michaelklishin** [reply]
RabbitMQ does not log either on start. The Docker image prints generated config file. See [User Seeding](https://www.rabbitmq.com/access-control.html#seeding) to find out what your other options are if there is no way to avoid this in the image.

**Comment by michaelklishin** [comment]
I no longer understand what this question is about but it seems to be Docker image-specific. [User Seeding](https://www.rabbitmq.com/access-control.html#seeding) explains what your options are. The most pragmatic and secure option is using definition export and [import on node boot](https://www.rabbitmq.com/definitions.html#import-on-boot).

Environment variable support in the Docker image is going away in the next few weeks when 3.9 previews begin shipping https://github.com/docker-library/rabbitmq/pull/467
